articles/151_node_to_participant_mapping.md

```diff
@@ -0,0 +1,213 @@
+{{ page.abstract }}
+</div>
+
+Original Author: {{ page.author }}
+
+## Introduction
+
+### What is a `Node`?
+
+In ROS, a `Node` is an entity used to group other entities.
+For example: `Publishers`, `Subscriptions`, `Services`, `Clients`.
+`Nodes` ease organization and code reusage, as they can be composed or launched in different ways.
+
+### What is a `Domain Participant`?
+
+A `Domain Participant` is a type of DDS entity.
+`Participants` also group other entities, like `Publishers`, `Subscribers`, `Data Writters`, `Data Readers`, etc.
+But participants do more than that:
+
+- Each `Participant` does discovery by its own.
+  Creating more than one `Participant` increase cpu and network.
+- Each `Participant` keeps track of the others and of each DDS entity.
+  Using more than one will duplicate that data.
+- Each `Participant` creates multiple threads for event handling, discovery, etc.
+  Creating more than one will increase cpu and memory usage.
+
+For those reasons, `Participants` are heavyweight.
+
+### Current status
+
+There is a one to one map between `Nodes` and `DDS Participants`.
+This simplified the design, as `DDS Participants` provide the same organization that a `Node` needs.
+The drawback of this approach, is that performance is deteriorated.
+Furthermore, the number of `Domain participants` is limited to a small number.
+For example, [RTI connext](https://community.rti.com/kb/what-maximum-number-participants-domain) is limited to 120 participants per domain.
+
+## Proposal
+
+The goal of this proposal is to improve overall performance by avoiding the creation of one `Domain Participant` per `Node`.
+API changes will be avoided, if possible.
+
+### What is a participant mapped to?
+
+There are two main alternatives:
+- One participant per process.
+- One participant per context.
+
+The second approach allows more flexibility.
+Considering that by default there's only one context per process, it won't lower the performance.
+Moreover, a mechanism for using the same participant in two context could be added.
+
+### What is a Node now?
+
+There's not lightweight DDS entity similar to a ROS `Node`, so `Nodes` have to be built from scratch.
+A `Node` should be able to:
+- Create other entities as `Publishers`, `Subscriptions`, `Services` and `Clients`.
+  `Nodes` should own those entities, that is to say, those entity shouldn't outlive a `Node`.
+- List all its entities.
+
+For all the entities, it should be possible to get the `Node` what created them.
+Each `Participant` should store all the information needed about its nodes, and communicate it to the others.
+
+### How `Node` information is communicated?
+
+#### Using a topic
+
+The name of all the available `Nodes`, and its `Publishers`, `Subscriptions`, `Services`, `Clients` should be available for every `Participant`.
+This information can be communicated using a `topic`.
+That topic will be an implementation detail and hidden to the user.
```

Comment:
@ivanpauno what does hidden to the user mean? Hidden by convention (i.e. leading underscore), hidden by not abiding to the ROS convention for DDS topic names (i.e. without an `rt/` prefix) ?

Reply:
The second one, I will be more explicit.